pushapkscript/test/test_script.py

```diff
@@ -92,14 +93,88 @@ def test_get_product_config_unknown_product():
 def test_get_product_config():
     context = Context()
     context.config = {
-        'products': {
-            'fenix': {
+        'products': [{
+            'product_names': ['fenix'],
+            'foo': 'bar',
+        }]
+    }
+
+    assert _get_product_config(context, 'fenix') == {'product_names': ['fenix'], 'foo': 'bar'}
+
+
+def test_get_publish_config_fennec():
+    aurora_config = {
+        'map_channels_to_apps': True,
+        'use_scope_for_channel': True,
+        'apps': {
+            'aurora': {
                 'foo': 'bar'
             }
         }
     }
+    assert _get_publish_config(aurora_config, {}, 'aurora') == {'foo': 'bar'}
+
 
-    assert _get_product_config(context, 'fenix') == {'foo': 'bar'}
+def test_get_publish_config_fennec_track_override():
+    aurora_config = {
+        'map_channels_to_apps': True,
+        'use_scope_for_channel': True,
+        'apps': {
+            'aurora': {
+                'foo': 'bar'
+            }
+        }
+    }
+    assert _get_publish_config(aurora_config, {'google_play_track': 'internal_qa'}, 'aurora') == {
+        'foo': 'bar',
+        'google_play_track': 'internal_qa',
+    }
+
+
+def test_get_publish_config_focus_old():
+    focus_config = {
+        'single_app_config': {
+            'foo': 'bar'
+        }
+    }
+    focus_payload = {'google_play_track': 'beta'}
+    assert _get_publish_config(focus_config, focus_payload, 'focus') == {'foo': 'bar', 'google_play_track': 'beta'}
+
+
+def test_get_publish_config_focus():
+    focus_config = {
+        'single_app_config': {
+            'foo': 'bar'
+        }
+    }
+    focus_payload = {'channel': 'beta'}
+    assert _get_publish_config(focus_config, focus_payload, 'focus') == {'foo': 'bar', 'google_play_track': 'beta'}
+
+
+def test_get_publish_config_fenix_old():
+    fenix_config = {
+        'map_channels_to_apps': True,
+        'apps': {
+            'nightly': {
+                'foo': 'bar',
+            }
+        }
+    }
+    focus_payload = {'google_play_track': 'nightly'}
+    assert _get_publish_config(fenix_config, focus_payload, 'focus') == {'foo': 'bar'}
+
+
+def test_get_publish_config_fenix():
+    fenix_config = {
+        'map_channels_to_apps': True,
```

Comment:
Nit: Let's add a test where `'map_channels_to_apps': False,` and `'use_scope_for_channel': True,` to simulate a bad config.